pkg/services/libraryelements/database.go

```diff
@@ -441,6 +441,42 @@ func (l *LibraryElementService) getAllLibraryElements(c context.Context, signedI
 	if folderFilter.parseError != nil {
 		return model.LibraryElementSearchResult{}, folderFilter.parseError
 	}
+
+	var foldersWithMatchingTitles []string
```

Comment:
We could extract all the following code to a function that returns these `foldersWithMatchingTitles` and keep the main function cleaner for better readability.